### PR TITLE
changes to lambdify (Fixes #9122) related to PR #9180

### LINKED ISSUE #9122: lambdifying with numpy module does not give an adequate traceback when arguments are missing
Hi I am experiencing a Bug when evaluating a lambdified matrix

``` python
B = sympy.lambdify(vsymbols, B, "numpy")
```

B is a large sparse matrix with a few non-zero elements which are square roots of symbolic variables.

I lambdify it to substitute numerical values a do some numeric computations more efficiently that with evalf. but I keep geting the error.

any ideas?

### PR DESCRIPTION
**Pull request by meghana-madhyastha**
I have created a separate pull request with my changes to the lambdify arguments as requested in PR #9180. I've made some changes based on the suggestions given there.If this is good to go, I will add the tests.

### CODE REVIEW

**Comment by bjodah** [comment]
Yes, please add some tests which highlights the purpose of these changes.

**Comment by meghana1995** [comment]
I've added the tests in context manager form as suggested

**Comment by bjodah** [comment]
Hmm.. this is a quite special case, I am surprised strings are even handled by lambdify, I can find a test using a comma separated string of variables, but I fail to find where this is documented. Have I missed it somewhere?

I would like to deprecate that behaviour: having different behaviour for different types passed to arguments is a recipe for unintuitive bahviour and messy code in my humble oppinion. I would recommend that users use `parse_expr` on strings before passing the to e.g. lambdify. Maybe
a dedicated convenience function would be motivated, e.g. `lambdify_str` if chaining `parse_expr` and `lambdify` turns out to be non-trivial.

As for this PR in particular I guess it solves the problem but someone who uses `lambdify` in that way will have to chime in.

File: sympy/utilities/tests/test_lambdify.py [hunk 1]
```diff
@@ -577,3 +577,11 @@ def test_issue_2790():
     assert lambdify((x, (y, z)), x + y)(1, (2, 4)) == 3
     assert lambdify((x, (y, (w, z))), w + x + y + z)(1, (2, (3, 4))) == 10
     assert lambdify(x, x + 1, dummify=False)(1) == 2
+
+def test_incorrect_arg_expr():
+    with raises(ValueError):
+        lambdify(x, "numpy")
```
Comment (smichr) [inline sympy/utilities/tests/test_lambdify.py:583]: Currently, this give the numpy module for any single input, `lambdify(x, "numpy")(1) -> numpy module`

File: sympy/utilities/tests/test_lambdify.py [hunk 2]
```diff
@@ -577,3 +577,11 @@ def test_issue_2790():
...
+
+def test_incorrect_arg_args():
+    with raises(ValueError):
+        lambdify("numpy", x)
```
Comment (smichr) [inline sympy/utilities/tests/test_lambdify.py:587]: Currently, this gives the symbol `x` for any single input, `lambdify('numpy', x) -> x`

**Comment by czgdp1807** [comment]
Closing this for now. However, please feel free to re-open it. Thanks for your contributions.

**Pull request closed without merging**